app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java

```diff
@@ -111,11 +135,27 @@ public void onFailure(@NonNull Exception e) {
                             }).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
                                                         @Override
                                                         public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
-                                                            dbRef.setValue(
+                                                            itemRef.setValue(
                                                                     new Item(
                                                                             categoryString, Float.parseFloat(priceString), description, taskSnapshot.getMetadata().getDownloadUrl().toString(), Integer.parseInt(stockString)
                                                                     )
                                                             );
+
+                                                            categoryRef.runTransaction(new Transaction.Handler() {
+                                                                @Override
+                                                                public Transaction.Result doTransaction(MutableData mutableData) {
+                                                                    HashMap<String, String> hashMap = (HashMap<String,String>) mutableData.getValue();
+                                                                    if (!hashMap.containsValue(categoryString)){
+                                                                        categoryRef.push().setValue(categoryString);
+                                                                    };
+                                                                    return null;
+                                                                }
+
+                                                                @Override
+                                                                public void onComplete(DatabaseError databaseError, boolean b, DataSnapshot dataSnapshot) {
+
+                                                                }
+                                                            });
                                                             //add shop id to item record
```

Comment:
This part is basically to add categories to the general category table. If it already exists then dont add.